File: tests-frontend/src/app/model/question.ts
import { IAnswer } from './answer';
import { IConcept, initConcept } from './concept';

export interface IQuestion {
  id: number | null;
  question: string;
  points: number;
  questionNumber: number;
  answers: IAnswer[];
  concept?: IConcept;
}

export const initQuestion: IQuestion = {
  id: null,
  question: '',
  points: 0,
  questionNumber: 0,
  answers: [],
  concept: initConcept,
};

File: tests-backend/sotis-project/src/main/java/com/example/sotisproject/model/Question.java
package com.example.sotisproject.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.beans.factory.annotation.Value;

import javax.persistence.*;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String question;

    @Column(columnDefinition = "integer default 10")
    private Integer points;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.EAGER)
    private Test test;

    @OneToMany(mappedBy = "question", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Set<Answer> answers;

    @ManyToOne(fetch = FetchType.EAGER)
    private Concept concept;

}
